src/main/java/org/genomicsdb/spark/sources/GenomicsDBPartitionReaderFactory.java

```diff
@@ -0,0 +1,24 @@
+package org.genomicsdb.spark.sources;
+
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.connector.read.InputPartition;
+import org.apache.spark.sql.connector.read.PartitionReader;
+import org.apache.spark.sql.connector.read.PartitionReaderFactory;
+import org.apache.spark.sql.types.StructType;
+
+import java.io.FileNotFoundException;
+import java.net.URISyntaxException;
+
+/**
+ * Creates an instance of PartitionReader for every partition.
+ **/
+public class GenomicsDBPartitionReaderFactory implements PartitionReaderFactory{
+ 
+  public GenomicsDBPartitionReaderFactory(){}
+
+  @Override
+  public PartitionReader<InternalRow> createReader(InputPartition partition){
+    return new GenomicsDBInputPartitionReader((GenomicsDBInputPartition) partition);
+  }
```

Comment:
Very likely I'm missing something - but creating a factory for this functionality seems to be an overkill.

Reply:
This is only an expectation from Spark a class that extends PartitionReaderFactory with functionality to return a PartitionReader.